swan-lake/integration-technology-tools/edi-tool.md

```diff
@@ -6,4 +6,349 @@ keywords: ballerina, programming language, edi, contract
+
+## Example
+
+A simple EDI schema (`edi-schema.json` file) is shown below.
+
+```json
+{
+    "name": "SimpleOrder",
+    "delimiters" : {"segment" : "~", "field" : "*"},
+    "segments" : {
+        "HDR": {
+            "tag" : "header",
+            "fields" : [{"tag" : "code"}, {"tag" : "orderId"}, {"tag" : "organization"}, {"tag" : "date"}]
+        },
+        "ITM": {
+            "tag" : "items",
+            "maxOccurances" : -1,
+            "fields" : [{"tag" : "code"}, {"tag" : "item"}, {"tag" : "quantity", "dataType" : "int"}]
+        }
+    }
+}
+```
+
+The above schema can be used to parse EDI documents with one HDR segment (mapped to `header`) and any number of ITM segments (mapped to `items`). The HDR segment contains three fields, which are mapped to `orderId`, `organization`, and `date`. Each ITM segment contains two fields mapped to `item` and `quantity`. Below is a sample EDI document that can be parsed using the above schema (`edi-sample1.edi` file).
+
+```
+HDR*ORDER_1201*ABC_Store*2008-01-01~
+ITM*A-250*12
+ITM*A-45*100
+ITM*D-10*58
+ITM*K-80*250
+ITM*T-46*28
+```
+
+### Generate Ballerina records
+
+Ballerina records for the EDI schema in the `edi-schema.json` file can be generated as follows (the generated Ballerina records will be saved in `orderRecords.bal`).
+
+```
+bal edi codegen -s resources/edi-schema1.json -o modules/hmartOrder/orderRecords.bal
+```
+
+The Ballerina records generated for the above schema are shown below.
+
+```ballerina
+type Header_Type record {|
+   string code?;
+   string orderId?;
+   string organization?;
+   string date?;
+|};
+
+type Items_Type record {|
+   string code?;
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+### Read EDI files
+
+The generated `fromEdiString` function can be used to read EDI text files into the generated Ballerina record as shown below. 
+
+>**Note:** Any data item in the EDI can be accessed using the record's fields as shown in the example code below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    string ediText = check io:fileReadString("resources/edi-sample1.edi");
+    SimpleOrder order1 = check hmartOrder:fromEdiString(ediText);
+    io:println(order1.header.date);
+}
+```
+
+### Write EDI files
+
+The generated `toEdiString` function can be used to serialize the `SimpleOrder` records into EDI text as shown below.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    SimpleOrder order2 = {...};
```

Comment:
why name the varilable as `order2`. Give a better name.

Reply:
Also can we have proper record instead of `{...}`

Reply:
@praneesha pls use the below initialization code:

SimpleOrder salesOrder = {header: {code: "HDR", orderId: "ORDER_200", organization: "HMart", date: "17-05-2023"}};
    salesOrder.items.push({code: "ITM", item: "A680", quantity: 15}); 
    salesOrder.items.push({code: "ITM", item: "A530", quantity: 2}); 
    salesOrder.items.push({code: "ITM",item: "A500", quantity: 4});

Reply:
Output of the above code is:

HDR*ORDER_200*HMart*17-05-2023~
ITM*A680*15~
ITM*A530*2~
ITM*A500*4~

Reply:
Updated the variable name to `salesOrder`.

Reply:
Also, updated the above codes.